GREEDY/ImagePreparation/Deskew.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GREEDY.ImagePreparation
+{
+    class Deskew
+    {
+        // Representation of a line in the image.  
+        private class HougLine
+        {
+            // Count of points in the line.
+            public int Count;
+            // Index in Matrix.
+            public int Index;
+            // The line is represented as all x,y that solve y*cos(alpha)-x*sin(alpha)=d
+            public double Alpha;
+        }
+
+
+        // The Bitmap
+        Bitmap _internalBmp;
+
+        // The range of angles to search for lines
+        const double ALPHA_START = -20;
+        const double ALPHA_STEP = 0.2;
+        const int STEPS = 40 * 5;
+        const double STEP = 1;
+
+        // Precalculation of sin and cos.
+        double[] _sinA;
+        double[] _cosA;
+
+        // Range of d
+        double _min;
+
+
+        int _count;
+        // Count of points that fit in a line.
+        int[] _hMatrix;
+
+
+        private Bitmap RotateImage(Bitmap bmp, float angle)
```

Comment:
why you need to rotate image if you did it before? or maybe you can use the same class if you really need to rotate at this point